### PR TITLE
Feature/new syscall tests cases

### PR DESCRIPTION
**Pull request by kpan2034**
Added tests for the following syscalls:

- pipe2
- getifaddrs
- getppid
- mkdir
- rmdir
- access

### CODE REVIEW

**Comment by rennergade** [comment]
We should hook these up to the test suite before we merge them. I'll let the team review the individual tests for now.

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: tests/test_cases/getppid.c
Comment (rennergade) [reply tests/test_cases/getppid.c:1]: Yes definitely.

**Comment by rennergade** [comment]
Also naming the test cases that handle two syscalls so we know that off the bat would be good. IE "access" should be "creat_access"

File: tests/test_cases/getppid.c
Comment (kpan2034) [reply tests/test_cases/getppid.c:1]: Alright, will work on this.

Comment (RusherRG) [reply tests/test_cases/getppid.c:1]: Awesome, this looks good. :tada:

File: tests/test_cases/pipe2.py
Comment (RusherRG) [inline tests/test_cases/pipe2.py:27]: Make sure to end all your test files with a new line.

File: tests/test_cases/getifaddrs.py
Comment (RusherRG) [inline tests/test_cases/getifaddrs.py:34]: Make sure to end all your test files with a new line.

File: tests/test_cases/getppid.c
Comment (RusherRG) [inline tests/test_cases/getppid.c:43]: Make sure to end all your test files with a new line.

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: tests/test_cases/pipe2.py
Comment (RusherRG) [inline tests/test_cases/pipe2.py:1]: Can you also add checks for the return value of read and write? Both of them should be 3(number of bytes read/write)

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: tests/test_cases/getppid.c
Comment (kuzeyardabulut) [inline tests/test_cases/getppid.c:14]: Why we are using `fflush()` in here ?

**Comment by kuzeyardabulut** [comment]
You should `fflush(stdout)` after doing printfs.

**Review by kuzeyardabulut — CHANGES_REQUESTED** [review]
You should `fflush(stdout)` after doing printfs.

File: tests/test_cases/pipe2.py
Comment (rennergade) [inline tests/test_cases/pipe2.py:34]: needs newline

File: tests/test_cases/pipe2.py
Comment (rennergade) [reply tests/test_cases/pipe2.py:1]: that would be good

File: tests/test_cases/pipe2.c
Comment (rennergade) [inline tests/test_cases/pipe2.c:32]: add fflushes

File: tests/test_cases/creat_access.c
Comment (kuzeyardabulut) [inline tests/test_cases/creat_access.c:44]: Do we need this?

File: tests/test_cases/getppid.c
Comment (kuzeyardabulut) [inline tests/test_cases/getppid.c:16]: I think it would be better to delete such short comment lines.

**Review by kuzeyardabulut — CHANGES_REQUESTED** [review]

**Pull request merged**
